Model/Api/GetProduct.php

```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
+            $parentProductInventory->setProduct($parentProduct);
+            $parentProductInventory->setStock($this->getStockStatus($parentProduct));
+            $this->productData->setParent($parentProductInventory);
+
+
+            $children = $parentProduct->getTypeInstance()->getUsedProducts($parentProduct);
+            $childrenStockArray = array();
+            foreach ($children  as $child) {
+                $childProductInventory = new ProductInventoryInfo();
+                $childProductInventory->setProduct($child);
+                $childProductInventory->setStock($this->getStockStatus($child));
+                array_push($childrenStockArray, $childProductInventory);
+            }
+            $this->productData->setChildren($childrenStockArray);
+        } elseif ($product->getTypeId() == "configurable") {
+            $children = $product->getTypeInstance()->getUsedProducts($product);
+            $childrenStockArray = array();
+            foreach ($children  as $child) {
+                $childProductInventory = new ProductInventoryInfo();
+                $childProductInventory->setProduct($child);
+                $childProductInventory->setStock($this->getStockStatus($child));
+                array_push($childrenStockArray, $childProductInventory);
+            }
+            $this->productData->setChildren($childrenStockArray);
+        }
     }
 
     // TODO: ADD unit tests @ethan
     /**
-     * Get user account associated with email
+     * Get product, its stock, and product family
      *
      * @api
      *
      * @param string $productID
+     * @param string $sku
      *
      * @return \Bolt\Boltpay\Api\Data\GetProductDataInterface
      *
      * @throws NoSuchEntityException
      * @throws WebapiException
      */
-    public function execute($productID = '')
+    public function execute($productID = '', $sku = '')
     {
-        if (!$this->hookHelper->verifyRequest()) {
-            throw new WebapiException(__('Request is not authenticated.'), 0, WebapiException::HTTP_UNAUTHORIZED);
-        }
-
-        if ($productID === '') {
-            throw new WebapiException(__('Missing product ID in the request parameters.'), 0, WebapiException::HTTP_BAD_REQUEST);
+        if ($productID === '' && $sku ==='') {
+            throw new WebapiException(__('Missing a product ID or a sku in the request parameters.'), 0, WebapiException::HTTP_BAD_REQUEST);
         }
 
         try {
-            $storeId = $this->storeManager->getStore()->getId();
-            $product = $this->productRepositoryInterface->getById($productID, false, $storeId, false);
-            $this->productData->setProduct($product);
-            $stockItem = $this->stockRegistry->getStockItem($product->getId());
-            $this->productData->setStock($stockItem);
+            $this->getProduct($productID, $sku);
+            $this->getProductFamily();
```

Comment:
These don't get or return anything, so an alternative naming would be better, like generate or collect